fsspec/spec.py

```diff
@@ -886,15 +886,13 @@ def get(self, rpath, lpath, recursive=False, callback=_DEFAULT_CALLBACK, **kwarg
             trailing_sep_maybe_asterisk,
         )
 
-        rpaths = self.expand_path(rpath, recursive=recursive)
-        if (
-            len(rpaths) == 1
-            and not recursive
-            and (trailing_sep(rpaths[0]) or self.isdir(rpaths[0]))
-        ):
-            # Non-recursive copy of directory does nothing.
-            return
         source_is_str = isinstance(rpath, str)
+        rpaths = self.expand_path(rpath, recursive=recursive)
+        if source_is_str and not recursive:
+            # Non-recursive glob does not copy directories
+            rpaths = [p for p in rpaths if not (trailing_sep(p) or self.isdir(p))]
```

Comment:
Alternative use of `filter` with a `lambda` here is slightly more code so I have opted for the easier to understand list comprehension.